src/d_main.c

```diff
@@ -1741,6 +1741,49 @@ void D_SetBloodColor(void)
   }
 }
 
+void D_SetTranslucency(int index)
+{
+  int i;
+
+  const int predefined_translucency[] = {
+    // MBF
+    MT_FIRE,      MT_SMOKE,     MT_FATSHOT,  MT_BRUISERSHOT,
+    MT_SPAWNFIRE, MT_TROOPSHOT, MT_HEADSHOT, MT_PLASMA,
+    MT_BFG,       MT_ARACHPLAZ, MT_PUFF,     MT_TFOG,
+    MT_IFOG,      MT_MISC12,    MT_INV,      MT_INS,
+    MT_MEGA,
+    // [Woof!]
+    MT_PLASMA1,   MT_PLASMA2
+  };
+
+  static boolean deh_set_translucency[arrlen(predefined_translucency)] = {false};
+
+  if (index)
+  {
+    for (i = 0; i < arrlen(predefined_translucency); ++i)
+    {
+      if (predefined_translucency[i] == index)
+      {
+        deh_set_translucency[i] = true;
+        break;
+      }
+    }
+
+    return;
+  }
+
+  for (i = 0; i < arrlen(predefined_translucency); ++i)
+  {
+    if (deh_set_translucency[i])
+      continue;
+
+    if (translucency || (strictmode && !demo_compatibility))
```

Comment:
So we still allow for translucency even in vanilla complevel? Good!

Reply:
Or was it meant to mean just that? Because now it means that translucency is unconditional enabled in strict mode for non-vanilla complevels. 🤔

Reply:
Yes, PrBoom+ enables it for complevel 9 and higher (comp_translucency)

Reply:
Technically you can always disable it with translucency percent = 100. Not sure if we need these checks.

Reply:
Yes, but then you *want* to play against the rules. You could also build your own executable...

Reply:
Will need to check with dsda, but don't the strict rules forbid translucency for vanilla complevels?

Reply:
Also, doesn't PrBoom+ only restrict the `general_translucency` config key to a specific complevel? I think removal of the predefined translucency flag is alway possible?

Reply:
User can always turn of translucency in PrBoom+. PrBoom+ enables predefined translucency in complevel 9 and higher.

In complevel vanilla we don't allow predefined translucency, but we do allow dehacked/PWAD translucency.

Reply:
If we want to copy PrBoom+/DSDA-Doom we need to move "Translucency for some things" into "Doom compatibility" menu and restore `general_translucency` in "General" menu.

Reply:
I don't want to copy PrBoom+ in every detail, just provide equivalent logic.

Reply:
> I don't want to copy PrBoom+ in every detail, just provide equivalent logic.

I think logic is equivalent, but PrBoom+ also has a `general_translucency` option to disable translucency for walls and things.